manim/animation/creation.py

```diff
@@ -668,3 +672,202 @@ def __init__(
+
+    Parameters
+    ----------
+    time_per_char
+        Frequency of appearance of the letters.
+    cursor
+        :class:`~.Mobject` shown after the last added letter.
+    buff
+        Controls how far away the cursor is to the right of the last added letter.
+    keep_cursor_y
+        If ``True``, the cursor's y-coordinate is set to the center of the ``Text`` and remains the same throughout the animation. Otherwise, it is set to the center of the last added letter.
+    leave_cursor_on
+        Whether to show the cursor after the animation.
+    kwargs
+        Additional arguments to be passed to the :class:`~.AddTextLetterByLetter` constructor.
+
+    .. tip::
+        This is currently only possible for class:`~.Text` and not for class:`~.MathTex`.
+
+
+    Examples
+    --------
+
+    .. manim:: InsertingTextExample
+        :ref_classes: Blink
+
+        class InsertingTextExample(Scene):
+            def construct(self):
+                text = Text("Inserting", color=PURPLE).scale(1.5).to_edge(LEFT)
+                cursor = Rectangle(
+                    color = GREY_A,
+                    fill_color = GREY_A,
+                    fill_opacity = 1.0,
+                    height = 1.1,
+                    width = 0.5,
+                ).move_to(text[0]) # Position the cursor
+
+                self.play(AddTextLetterByLetterWithCursor(text, cursor))
+                self.play(Blink(cursor, how_many_times=2))
+
+    """
+
+    def __init__(
+        self,
+        text: Text,
+        cursor: Mobject,
+        buff: float = 0.1,
+        keep_cursor_y: bool = True,
+        leave_cursor_on: bool = True,
+        suspend_mobject_updating: bool = False,
+        int_func: Callable[[np.ndarray], np.ndarray] = np.ceil,
+        rate_func: Callable[[float], float] = linear,
+        time_per_char: float = 0.1,
+        run_time: float | None = None,
+        reverse_rate_function=False,
+        introducer=True,
+        **kwargs,
+    ) -> None:
+        self.cursor = cursor
+        self.buff = buff
+        self.keep_cursor_y = keep_cursor_y
+        self.leave_cursor_on = leave_cursor_on
+        super().__init__(
+            text,
+            suspend_mobject_updating=suspend_mobject_updating,
+            int_func=int_func,
+            rate_func=rate_func,
+            time_per_char=time_per_char,
+            run_time=run_time,
+            reverse_rate_function=reverse_rate_function,
+            introducer=introducer,
+            **kwargs,
+        )
+
+    def begin(self) -> None:
+        self.y_cursor = self.cursor.get_y()
+        self.cursor.initial_position = self.mobject.get_center()
+        if self.keep_cursor_y:
+            self.cursor.set_y(self.y_cursor)
+
+        self.cursor.set_opacity(0)
+        self.mobject.add(self.cursor)
+        super().begin()
+
+    def finish(self) -> None:
+        if self.leave_cursor_on:
+            self.cursor.set_opacity(1)
+        else:
```

Comment:
Does the else actually do anything? I tried deleting the else statement and the behavior was still the same (although I might be wrong).

Reply:
It's been a while so I don't really remember why this is.
But here's an example scene I used to test the animations
```python
class Typing(Scene):
    def construct(self):
        text = Text("Typing", color=PURPLE).scale(1.5).to_edge(LEFT)
        cursor = Rectangle(
            color = GREY_A,
            fill_color = GREY_A,
            fill_opacity = 1.0,
            height = 1.1,
            width = 0.5,
        ).move_to(text[0]) # Position the cursor

        self.play(Blink(cursor, how_many_times=2))
        self.play(AddTextLetterByLetterWithCursor(text, cursor, leave_cursor_on=False)) # Turning off the cursor is important
        self.play(Blink(cursor, how_many_times=2))
        self.play(RemoveTextLetterByLetterWithCursor(text, cursor))
        self.play(Blink(cursor, how_many_times=2, ends_with_off=True))
        return super().construct()
```

The output WITH the else

https://github.com/ManimCommunity/manim/assets/111203113/851f5389-5514-4b28-8094-ed8f2c84ac4e

The output WITHOUT the else

https://github.com/ManimCommunity/manim/assets/111203113/9d177f43-991b-4de8-bed4-4ac6460888e5

Reply:
Weird, I had the same output for both.
Since the else seems to fix the issue on your computer, I don't see a need to remove it, so I'll just resolve this comment.